src/sgame/sg_votes.cpp

```diff
@@ -0,0 +1,1340 @@
+		trap_SendServerCommand( -1, cmd );
+	}
+	else
+	{
+		G_TeamCommand( team, cmd );
+	}
+
+	G_ResetVote( team );
+}
+
+void G_ResetVote( team_t team )
+{
+	int i;
+
+	level.team[ team ].voteTime = 0;
+	level.team[ team ].voteYes = 0;
+	level.team[ team ].voteNo = 0;
+	level.team[ team ].voted = 0;
+
+	for ( i = 0; i < level.maxclients; i++ )
+	{
+		level.clients[ i ].pers.voted &= ~( 1 << team );
+		level.clients[ i ].pers.voteYes &= ~( 1 << team );
+		level.clients[ i ].pers.voteNo &= ~( 1 << team );
+	}
+
+	trap_SetConfigstring( CS_VOTE_TIME + team, "" );
+	trap_SetConfigstring( CS_VOTE_STRING + team, "" );
+	trap_SetConfigstring( CS_VOTE_YES + team, "0" );
+	trap_SetConfigstring( CS_VOTE_NO + team, "0" );
+}
+
+/*
+==================
+G_Vote
+==================
+*/
+void G_Vote( gentity_t* ent, team_t team, bool voting )
+{
+	if ( !level.team[ team ].voteTime )
+	{
+		return;
+	}
+
+	if ( voting && ( ent->client->pers.voted & ( 1 << team ) ) )
+	{
+		return;
+	}
+
+	if ( !voting && !( ent->client->pers.voted & ( 1 << team ) ) )
+	{
+		return;
+	}
+
+	ent->client->pers.voted |= 1 << team;
+
+	// TODO maybe refactor vote no/yes in one and only one variable and divide the NLOC by 2 ?
+
+	if ( voting )
+	{
+		level.team[ team ].voted++;
+	}
+	else
+	{
+		level.team[ team ].voted--;
+	}
+
+	if ( ent->client->pers.voteYes & ( 1 << team ) )
+	{
+		if ( voting )
+		{
+			level.team[ team ].voteYes++;
+		}
+		else
+		{
+			level.team[ team ].voteYes--;
+		}
+
+		trap_SetConfigstring( CS_VOTE_YES + team, va( "%d", level.team[ team ].voteYes ) );
+	}
+
+	if ( ent->client->pers.voteNo & ( 1 << team ) )
+	{
+		if ( voting )
+		{
+			level.team[ team ].voteNo++;
+		}
+		else
+		{
+			level.team[ team ].voteNo--;
+		}
+
+		trap_SetConfigstring( CS_VOTE_NO + team, va( "%d", level.team[ team ].voteNo ) );
+	}
+}
+
+static constexpr char MARKER = '$';
+
+static bool IsEscapedMarker( Str::StringRef s, size_t pos )
+{
+	return pos + 1 < s.size() && s[ pos ] == MARKER && s[ pos + 1 ] == MARKER;
+}
+
+// Use a custom escape to avoid issues with Cmd::Escape unintentionally terminating quotes
+// for strings based on user input.
+static std::string VoteEscape( Str::StringRef s, bool escape )
+{
+	if ( !escape )
+	{
+		return s;
+	}
+	std::string out;
+	out.reserve( s.size() * 2 );
+	for ( const auto& c : s )
+	{
+		out += '\\';
+		out += c;
+	}
+	return out;
+}
+
+// We only want to escape for commands that are executed. For display only, there is no injection concern,
+// so don't escape those.
+static std::string G_HandleVoteTemplate( Str::StringRef str, gentity_t* ent, team_t team,
+                                         std::string& cmd, std::string& arg, std::string& reason,
+                                         std::string& name, int clientNum, int id, bool escape )
+{
+	std::unordered_map<std::string, std::string> params = {
+		{ "team", BG_TeamNamePlural( team ) },
+		{ "arg", VoteEscape( arg, escape ) },
+		{ "reason", VoteEscape( reason, escape ) },
```

Comment:
You should either add a way to set reasonNeeded or get rid of `$reason$`

Reply:
done.